File: URI/Lua/ParImparOuRoubo.lua

p = io.read("*number")
j1 = io.read("*number")
j2 = io.read("*number")
r = io.read("*number")
a = io.read("*number")
if (r == 1 and a == 1) then
    print("Jogador 2 ganha!")
elseif ( (p == 1 and (j1+j2)%2 == 0) or (p == 0 and (j1+j2)%2 == 1) or (r == 1 and a == 0)) then
    print("Jogador 1 ganha!")
else
    print("Jogador 2 ganha!")
end
    

File: URI/C/ParImparOuRoubo.c
#include <string.h>
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
// macros 
#define showi(x) printf("%s -> %d\n", #x, x);
#define showf(x) printf("%s -> %f\n", #x, x);
#define shows(x) printf("%s -> %s\n", #x, x);
#define showd(x) printf("%s -> %lf\n", #x, x);
// typedefs
typedef unsigned int uint;
typedef long int lint;
typedef long long int llint;
typedef unsigned long long int ulli;
// bool simulation
#define true 1
#define false 0

int main(void){
    int p, j1, j2, r, a;
    scanf("%d %d %d %d %d", &p, &j1, &j2, &r, &a);
    if (r == 1 && a == 1)
    {
        printf("Jogador 2 ganha!");
    }
    else if ( (p == 1 && (j1+j2)%2 == 0) || (p == 0 && (j1+j2)%2 == 1) || (r == 1 && a == 0))
    {
        printf("Jogador 1 ganha!");
    }
    else
    {
        printf("Jogador 2 ganha!");
    }
    putchar('\n');
    
    
    return 0;
}